### ISSUE TITLE
Should the suppression operator affect the output value of a ref argument

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#35555]
```C#
string s = "";
M(ref s!); // where void M(ref string? x)
// should there be a warning for assignment from parameter x to argument s? Currently we do not produce that warning
s.ToString(); // what is the state of s? Currently it is non-null
```

In the current definition of the operator, there are two effects: change the state of an expression (`s` in this example) to non-null, and suppression conversion warnings related to nullability.

Relates to https://github.com/dotnet/roslyn/pull/35538#pullrequestreview-234647721

**Comment by jcouv** [comment]
From discussion with Mads and Andy today, this behavior is fine.
